src/server/bitops_family.cc

```diff
@@ -159,7 +241,49 @@ void BitPos(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void BitCount(CmdArgList args, ConnectionContext* cntx) {
-  (*cntx)->SendLong(0);
+  // Support for the command BITCOUNT
+  // See details at https://redis.io/commands/bitcount/
+  // Please note that if the key don't exists, it would return 0
+
+  if (args.size() == 3 || args.size() > 5) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  // return (*cntx)->SendLong(0);
+  std::string_view key = ArgS(args, 1);
+  bool as_bit = false;
+  std::optional<int32_t> start = std::nullopt;
```

Comment:
i think using optional here is possible but not required. You can use default 0 for start and default UINT32_MAX for end. this would have the same impact. Also, consider using uint64_t or size_t for these because bit indices can easily go across 4B.

Reply:
You're right about the 64 bits. I will remove the optional (though I think its a correct use since these are really optional values)

Reply:
do you think size_t < uint64_t - normally they should have == sizeof?